Automattic/woocommerce-payments
# Add Compatibility Service

#### Changes proposed in this Pull Request

This PR adds the Compatibility Service that will allow us to check what version of WooCommerce core is being used alongside WooPayments for faster updates when there may be compatibility issues between versions.

* Add the `Compatibility_Service` class in the `WCPay` namespace.
  * The primary method here is `update_compatibility_data`, which is called by the `woocommerce_payments_account_refreshed` action. This action is called when the account object is refreshed on the client side, so every 2 hours if admins are active, or 24 hours if only customers are active.
  * The `update_compatibility_data` method will call the `WC_Payments_API_Client` method with the same name. It passes along an array that includes the WooCommerce core version, or a notice if the constant is not defined.
* Add the `update_compatibility_data` method to `WC_Payments_API_Client`. This handles POSTing the data passed to it to the `COMPATIBILITY_API` endpoint on the server.
* The test `get_onboarding_po_eligible` was missing `test_` as the prefix, which caused it not to run, so I updated that.
* Added tests to cover all changes.

#### Testing instructions

* Please see https://github.com/Automattic/woocommerce-payments-server/pull/4222 to test

-------------------


**Post merge**
